include/zephyr/net/phy.h

```diff
@@ -99,6 +99,10 @@ typedef void (*phy_callback_t)(const struct device *dev,
  * public documentation.
  */
 __subsystem struct ethphy_driver_api {
+	/** Get MAC address */
+	int (*get_mac_address)(const struct device *dev, uint8_t port,
```

Comment:
There's nothing in this driver that has a notion of a `port`, is this needed?

Reply:
Well, lets consider the KSZ9897, or DM8806 PHY's which will use this API. They have six PHY's inside, which can control e.g.  5 Ethernet ports. This parameter is here tell the driver from which PHY correlate usually with Ethernet port we want to get the MAC address. Which will be in fact the MAC address of connected device.

Reply:
I'm starting to think you need to take a look at the DSA driver `include/zephyr/net/dsa.h`

Reply:
Hmm, `dsa_switch_get_mac_table_entry()` looks similar let me think....

Reply:
> I'm starting to think you need to take a look at the DSA driver `include/zephyr/net/dsa.h`

I was going to say the same thing